### ISSUE TITLE
Installing Cypress using a private registry (Artifactory)

### ISSUE THREAD

**Issue by kobimantzur** [cypress-io/cypress#6877]
### Current behavior:

Cypress install fails when serving it from a private registry (Artifactory). 
The package relies on the version of "request" that exists on github/cypress-io instead of npmjs latest version.

We have no option to whitelist Github on the server that tries to install Cypress. 

```
npm WARN applications@1.0.0 requires a peer of jest@24.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN applications@1.0.0 No description
npm WARN applications@1.0.0 No repository field.

npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/cypress-io/request.git
npm ERR!
npm ERR! ssh: connect to host github.com port 22: Connection refused
npm ERR! fatal: Could not read from remote repository.
npm ERR!
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR!
npm ERR! exited with error code: 128

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kobi.mantzur/.npm/_logs/2020-03-29T10_30_12_999Z-debug.log
```

### Desired behavior:

Fetch Request from npmjs, it will allow us to store it in Artifactory like the rest of the dependencies of Cypress.

### Versions
Cypress Version: 4.2.0
OS: CentOS Linux 7 (Core)

**Comment by mbarzeev** [comment]
+1
Having a dependency on Github fork presents security risks.

**Comment by alonseg** [comment]
+1

**Comment by ariglasser** [comment]
+1  A must-have feature

**Comment by jennifer-shehane** [comment]
Duplicate of #6752. The fix for this will be released today in our 4.3.0 release.
